.changeset/odd-mugs-enjoy.md

```diff
@@ -0,0 +1,22 @@
+---
+'sku': major
+---
+
+Remove type-only imports during transpilation
+
+This change enables babel to mimic the behaviour of TypeScript's [`verbatimModuleSyntax`] compiler option. The following code demonstrates the result of this change when transpiling TypeScript to JavaScript:
+
+```ts
+// Erased entirely
+import type { A } from "a";
+
+// Rewritten to `import { b } from "bcd";`
+import { b, type c, type d } from "bcd";
+
+// Rewritten to `import {} from "xyz";`
+import { type xyz } from "xyz";
```

Comment:
This one surprised me, shouldnt it be erased entirely as well? Or is that incase there are side effect imports in `xyz`?

Reply:
Yep, it just strips the type import, leaving the side-effect. In practice though, our lint rules would convert this to `import type { xyz } from "xyz"`, so it actually _would_ be stripped entirely.